src/Native/System.Native/pal_runtimeinformation.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_runtimeinformation.h"
+#include "pal_types.h"
+#include <stdio.h>
+#include <sys/utsname.h>
+
+extern "C" const char* GetUnixVersion()
+{
+    struct utsname _utsname;
+    size_t length = _UTSNAME_LENGTH * 3 + 2;
+    char * version = new char[length];
+    if (uname(&_utsname) != -1)
+    {
+        snprintf(version, length, "%s %s %s", _utsname.sysname, _utsname.release, _utsname.version);
+    }
+
+    return version;
+}
+
+/* Returns an int representing the OS Architecture:
+ 0 - x86
+ 1 - x64
+ 2 - ARM */
+extern "C" int32_t GetUnixArchitecture()
+{
+#if defined(ARM)
+    return 2;
+#elif defined(BIT64)
+    return 1;
```

Comment:
Yes, for now, but when we add another 64-bit arch we'd set BIT64 there too. See a hint of the future in [coreclr repo](https://github.com/dotnet/coreclr/blob/194d5a41455bda3f348524a1b54f59b5e0c19f0c/src/pal/src/CMakeLists.txt#L65).